File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/text/MessageRenderer.java
package top.likoslupus.cellulosesz.api.text;

public interface MessageRenderer {

    default RichText render(String locale, String key) {
        return render(locale, key, MessageArguments.empty());
    }

    RichText render(
            String locale,
            String key,
            MessageArguments arguments
    );

    default RichText render(String locale, LocalizedMessage message) {
        return render(locale, message.key(), message.arguments());
    }

    default RichText renderInline(String locale, String template) {
        return renderInline(locale, template, MessageArguments.empty());
    }

    RichText renderInline(
            String locale,
            String template,
            MessageArguments arguments
    );

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/text/MessageRenderer.kt
package top.likoslupus.cellulosesz.api.text

public interface MessageRenderer {

    public fun render(
        locale: String,
        key: String
    ): RichText =
        render(
            locale,
            key,
            MessageArguments.empty()
        )

    public fun render(
        locale: String,
        key: String,
        arguments: MessageArguments
    ): RichText

    public fun render(
        locale: String,
        message: LocalizedMessage
    ): RichText =
        render(
            locale,
            message.key,
            message.arguments
        )

    public fun renderInline(
        locale: String,
        template: String
    ): RichText =
        renderInline(
            locale,
            template,
            MessageArguments.empty()
        )

    public fun renderInline(
        locale: String,
        template: String,
        arguments: MessageArguments
    ): RichText

}
